drivers/adc/adc_stm32.c

```diff
@@ -455,7 +455,8 @@ static void adc_stm32_disable(ADC_TypeDef *adc)
 	!defined(CONFIG_SOC_SERIES_STM32G0X) && \
 	!defined(CONFIG_SOC_SERIES_STM32L0X) && \
 	!defined(CONFIG_SOC_SERIES_STM32WBAX) && \
-	!defined(CONFIG_SOC_SERIES_STM32WLX)
+	!defined(CONFIG_SOC_SERIES_STM32WLX) && \
+	!defined(CONFIG_SOC_SERIES_STM32U0X)
```

Comment:
Keep the series in alphabetical order, this should go before WBA here. To repeat everywhere necessary.